ground_truths:
  drone_pose:
    name: drone_pose
    num_states: 6
    topic:
      sim: /ground_truth/pose/drone
      real: /qualisys/anafi/pose
    file_header: "Format: timestamp[sec] x[m], y[m], z[m], phi[deg], theta[deg], psi[deg]]\n"
    max_values_stored_in_buffer: 10

  helipad_pose:
    name: helipad_pose
    num_states: 6
    topic:
      sim: /ground_truth/pose/helipad
      real: /qualisys/helipad/pose
    file_header: "Format: timestamp[sec] x[m], y[m], z[m], phi[deg], theta[deg], psi[deg]]\n"
    max_values_stored_in_buffer: 10

estimates:
  dnn_cv_pose:
    name: dnn_cv_pose
    num_states: 4
    topic:
      sim: /estimate/drone_pose/dnnCV
      real: /estimate/drone_pose/dnnCV
    file_header: "Format: timestamp[sec] [x[m], y[m], z[m], psi[deg]]\n"
    max_values_stored_in_buffer: 10
  
  ekf_output:
    name: ekf_output
    num_states: 56 # 7 states + 7x7 covariance matrix
    topic:
      sim: /estimate/ekf
      real: /estimate/ekf
    file_header: "Format: timestamp[sec] [x_drone[m], y_drone[m], z_drone[m], 
                  psi_drone[deg], vx_helipad[m/s], vx_helipad[m/s], vx_helipad[m/s],
                  covariance for each state (flattened 7x7 matrix)]\n"
    max_values_stored_in_buffer: 10
